### PR TITLE
Add contributing guide.

### PR DESCRIPTION
**Pull request by bernardoamc**
This is a first stab at trying to add a `Contributing Guide` page to exercism. The motivation for this is that it would be good if people could have a centralized place to see where they can help exercism and be redirected to the relevant repository contributing guide sections.

Refer to this [discussion](https://github.com/exercism/discussions/issues/22) for more information.

## Current layout
screen shot 2016-05-31 at 10 24 15 pm
screen shot 2016-05-31 at 10 24 39 pm
screen shot 2016-05-31 at 10 24 56 pm

### CODE REVIEW

**Comment by bernardoamc** [comment]
There are a few sections that it would be awesome if someone could help me with the text, mainly the ones with `Section description here.` placeholder.

**Comment by bernardoamc** [comment]
Also, the page layout is pretty simple right now, is it ok as is?

**Comment by kytrinyx** [comment]
> There are a few sections that it would be awesome if someone could help me with the text

Yeah! I'll definitely help out with actual copy. For now I'm just going to post some thoughts/reactions.

> the page layout is pretty simple right now, is it ok as is?

I think it probably is... I have one idea that I want to throw out there, with the caveat that I don't know what I'm doing when it comes to visual stuff.

What if we had this sort of structure http://getbootstrap.com/2.3.2/examples/justified-nav.html ? The intro would be instead of the "marketing stuff" section, and then we could have individual sections below it in several columns.

We'd probably need to make up short titles for each way to help. Thinking out loud...

- "Make Up Exercises"
- "Translate Exercises"
- "Review Pull Requests"
- "Extract Canonical Data"
- "Help Triage Issues"
- "Fix Bugs"
- "Add Features"
- "Design Stuff"
- "Help Us Think"

For each section it would be neat if we could start off with the benefit--why this is important. Unless it's obvious, I suppose. Also, if this is something that people who are new to programming can tackle, it would be great to say so.

Another part of the intro that might be interesting to people is _what is important for Exercism right now?_. For the past year or so the important things have been:

- **Make it easy to understand what Exercism is and who it is for.** I rewrote the homepage. There's still work to do, but it's vastly better.
- **Improve the onboarding experience.** Make it so there are no dead ends, and help newer programmers not be overwhelmed by all the jargon and command-line stuff. I rewrote the "how it works" pages, and it's a bit better than it was, but could still be improved. We can link to this issue: https://github.com/exercism/discussions/issues/18
- **Improve the feedback loop** - not everyone gets feedback, not all feedback is helpful.
- **Make it easier to contribute** - things like this pull request and improved documentation, but also improving the codebase (which mostly is a complete and utter mess in the main website repository)

There are other things that we're working on, performance for example, but the above four things have been what I've been thinking about.

**Comment by bernardoamc** [comment]
> What if we had this sort of structure http://getbootstrap.com/2.3.2/examples/justified-nav.html ? The intro would be instead of the "marketing stuff" section, and then we could have individual sections below it in several columns.

I will rewrite the page using this format and we can see how it goes. :)

> Another part of the intro that might be interesting to people is what is important for Exercism right now?. 

This is great, I think the first step to improve contributions is for people to feel "safe" and understand the project goals.

**Comment by bernardoamc** [comment]
I think the header got a bit strange following that bootstrap example, but the sections are great!

## Header
screen shot 2016-05-30 at 9 21 34 pm

## Sections
screen shot 2016-05-30 at 9 22 13 pm

**Comment by bernardoamc** [comment]
Should we really track the generated `css` files?

**Comment by kytrinyx** [comment]
I love the new layout. I think this is definitely the way to go.

> I think the first step to improve contributions is for people to feel "safe" and understand the project goals.

Yepp, I think you're absolutely right.

> Should we really track the generated css files?

Technically, no, but until we rework everything to have the package.json at the root level, we need to, otherwise we have to figure out how to hack one of the heroku build packs. I haven't wanted to take the time to do the latter (and I don't like front-end work enough to convince myself to do the former, because i can see myself sinking a few days into it, and I don't really have a few days).

**Comment by bernardoamc** [comment]
> I love the new layout. I think this is definitely the way to go.

Awesome! This leaves us with filling the missing information and polishing the intro. ;)

> Technically, no, but until we rework everything to have the package.json at the root level, we need to, otherwise we have to figure out how to hack one of the heroku build packs. I haven't wanted to take the time to do the latter (and I don't like front-end work enough to convince myself to do the former, because i can see myself sinking a few days into it, and I don't really have a few days).

Fair enough! 👍  Maybe I can take a look into these tasks after this PR, but one step at a time for now, better to not rush things.

File: app/views/site/contribute.erb [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    <p class="lead">
+      <strong>So you want to contribute to exercism?</strong><br />
+      That is great news! We welcome all sort of contributions and will do everything we can to help you get started.
+    </p>
+
+    <p>
+      <strong>Exercism is a project comprised of several repositories and these can be divided basically in:</strong>
+
+      <ol>
+        <li><strong>The website</strong>: Its here that we can improve exercism features or layout.</li>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: This one is different from all of the others. All of the following bullet points explain what the thing does, whereas this one says how to contribute. How about saying something like:

_Browse and review solutions, and have conversations._

File: app/views/site/contribute.erb [hunk 2]
```diff
@@ -0,0 +1,130 @@
+<div class="container">
+  <section class="contributing-intro jumbotron">
+    <h1>Contributing Guide</h1>
+
+    <p class="lead">
+      <strong>So you want to contribute to exercism?</strong><br />
+      That is great news! We welcome all sort of contributions and will do everything we can to help you get started.
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: s/sort/sorts/

File: app/views/site/contribute.erb [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  <hr>
+
+  <section class="contributing-areas">
+    <div class="row">
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Make Up Exercises</h2>
+
+          <p>
+            If you have an idea for an exercise you can go to the <a href="https://github.com/exercism/x-common">x-common</a> repository, there you will find existing exercises and can check if your exercise is not already defined.
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: How about:

_We're always looking to add new exercises. Have an idea? Suggest it in the [metadata repository](https://github.com/exercism/x-common/issues)._

_Check out the section about [implementing a new exercise](https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md#implementing-a-completely-new-exercise) in the Contributing Guide._

links:
- https://github.com/exercism/x-common/issues
- https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md#implementing-a-completely-new-exercise

File: app/views/site/contribute.erb [hunk 4]
```diff
@@ -0,0 +1,130 @@
+<div class="container">
+  <section class="contributing-intro jumbotron">
+    <h1>Contributing Guide</h1>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Let's not call this the Contributing Guide, since that's the official name for the files that are called `CONTRIBUTING.md` (and we have two of them... which probably is a bad idea. hm.)

File: app/views/site/contribute.erb [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      <strong>So you want to contribute to exercism?</strong><br />
+      That is great news! We welcome all sort of contributions and will do everything we can to help you get started.
+    </p>
+
+    <p>
+      <strong>Exercism is a project comprised of several repositories and these can be divided basically in:</strong>
+
+      <ol>
+        <li><strong>The website</strong>: Its here that we can improve exercism features or layout.</li>
+        <li><strong>Command line interface</strong>: Fetch exercises and submit solutions.</li>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _Used to fetch exercises and submit solutions._

File: app/views/site/contribute.erb [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        <li><strong>Language track repositories</strong>: Each language has its own repository with available problems.</li>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Maybe:

_Contain the language-specific implementations of exercises. Each language has its own repository._

(Can you tell I'm trying to normalize the format of the sentences? I'm not quite there yet.)

File: app/views/site/contribute.erb [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        <li><strong>Problem metadata</strong>: Contains problem descriptions.</li>
+        <li><strong>Problems API</strong>: Delivers exercises, consumed by the Exercism command-line client, as well as the Exercism website.</li>
+        <li><strong>Rikki- the robot</strong>: Provides automated feedback on certain exercises.</li>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _Provides helpful, friendly, automated feedback on certain exercises._ -- trying to make it sound a bit less impersonal.

File: app/views/site/contribute.erb [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+            You can check <a href="https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md#implementing-a-completely-new-exercise">here</a> to learn how to implement a new exercise.
+          </p>
+        </div>
+      </div>
+
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Translate Exercises</h2>
+
+          <p>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Maybe something like...

_Another great way to add new exercises is port an existing exercise to one of the other languages. This is easier than making up an exercise from scratch, since other people have thought through which test cases are needed._

_Track maintainers are more than happy to give you all the support you need to successfully contribute._

File: app/views/site/contribute.erb [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+            Translating an exercise to your favorite language is a great way to help the comunity, track maintainers will be more than happy to give all the support needed so you can complete your first contribution.
+          </p>
+
+          <p>
+            A detailed guide can be seen <a href="https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md#porting-an-exercise-to-another-language-track">here</a> if you want to give it a try!
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: s/A detailed guide can be seen here/Check out the detailed guide/ -- active call to action, not passive

File: app/views/site/contribute.erb [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+          </p>
+        </div>
+      </div>
+
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Review Pull Requests</h2>
+
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _We want the quality of the exercises to be as high as possible (and reviewing pull requests is a great way to both learn and teach a programming language)._

_We look at all sorts of things: style, idiomatic use of language features, naming, missing test cases, and redundant test cases. We also try to avoid forcing the solution to take a particular approach. Subscribe to the language track repositories of your favorite languages so you get notifications about new issues and pull requests. The track maintainers will greatly appreciate your help!_

File: app/views/site/contribute.erb [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+          <p>
+            Reviewing Pull Requests is a great way to both learn and teach a programming language, the track maintainers will really appreciate your help!
+          </p>
+
+          <p>
+             You can search your language repository <a href="https://github.com/exercism/">here</a>.
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Nice!

Maybe _You can search for language track repositories here._

File: app/views/site/contribute.erb [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+          </p>
+        </div>
+
+      </div>
+    </div>
+
+    <div class="row">
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Canonical Data</h2>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Hm, now that I see this, it should be a call to action. Maybe _Extract_ canonical data?

File: app/views/site/contribute.erb (hunk 4 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: This works like a summary for multiple contributing guides, naming this will be tricky.

File: app/views/site/contribute.erb [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+
+          <p>What is this section about?</p>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _Different implementations of the same exercise often drift apart as people solving them discover different, interesting edge cases. Defining a canonical set of test data helps improve the quality all the implementations, and makes it easier to port the exercise to new languages._

_Do a survey of the existing implementations, and decide on a good set of test data. Submit a pull request to [the metadata repository](https://github.com/exercism/x-common) suggesting a good set of test data, and the track maintainers will be delighted to discuss it with you._

File: app/views/site/contribute.erb (hunk 12 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: `Extract canonical data` don't fit in a single line (but it is a better title), hmm...

File: app/views/site/contribute.erb [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+          <p>A link to further instructions.</p>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Ah, this is missing from the contributing guide. https://github.com/exercism/x-common/issues/261

File: app/views/site/contribute.erb (hunk 4 above)
Comment (kytrinyx) [inline app/views/site/contribute.erb]: How about "Contribute to Exercism"?

**Comment by kytrinyx** [comment]
> one step at a time for now, better to not rush things.

Completely agree!

File: app/views/site/contribute.erb (hunk 12 above)
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Ok, thinking...

`Survey Implementations`?

Comment (kytrinyx) [inline app/views/site/contribute.erb]: Or `Canonicalize Data`?

File: app/views/site/contribute.erb (hunk 14 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: Nice, when we merge this we can add the missing sections. :)

File: app/views/site/contribute.erb [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+        </div>
+      </div>
+
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Help Triage Issues</h2>
+
+          <p>What is this section about?</p>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _We've got a huge number of old issues that aren't particularly actionable, and new issues come in all the time._

_Is an issue no longer relevant? Can it be closed? Do we need more information? If it's a bug, is it reproducible? If it's a discussion and we need to make some sort of decision, then it should be labeled as such._

Maybe we can link to this discussion for now: https://github.com/exercism/discussions/issues/20

File: app/views/site/contribute.erb [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+          <p>A link to further instructions.</p>
+        </div>
+      </div>
+
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Fix Bugs</h2>
+
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _Exercism is currently a collection of experiments that grew into an app with the help of duct tape, chewing gum, and chicken wire. It's a mess, and it's got bugs._

- link to open bugs: https://github.com/exercism/exercism.io/issues?q=is%3Aissue+is%3Aopen+label%3Abug

File: app/views/site/contribute.erb [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+          <p>What is this section about?</p>
+          <p>A link to further instructions.</p>
+        </div>
+      </div>
+    </div>
+
+    <div class="row">
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Add Features</h2>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: I wonder if we should skip this section until we've got the roadmap a bit more under control. I don't know what to say here :)

File: app/views/site/contribute.erb [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+
+          <p>What is this section about?</p>
+          <p>A link to further instructions.</p>
+        </div>
+      </div>
+
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Design Stuff</h2>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _Exercism has for the most part been designed by accident by non-designers, and it shows!_

_We need help at every level of design. Know how to design icons? Wonderful. Want to make the pages look consistent and line things up? Fantastic. Want to redesign an entire page or feature or section? [Open an issue](https://github.com/exercism/discussions/issues) so we can talk through some of the hard-earned learning that we've picked up along the way._

File: app/views/site/contribute.erb [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+
+          <p>What is this section about?</p>
+          <p>A link to further instructions.</p>
+        </div>
+      </div>
+
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Help Us Think</h2>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _There's a lot that we haven't quite figured out about how to make Exercism as useful as possible._

_If you like to think through usability issues, communication, design, features, roadmap, collaboration, or any other exciting open source stuff, join us in the [discussions repository](https://github.com/exercism/discussions/issues), where we're trying to sort out Exercism's future._

File: app/views/site/contribute.erb (hunk 12 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: I would go with the last one!

File: app/views/site/contribute.erb (hunk 17 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: Maybe something like:

_Exercism is a collective work, we are always open to suggestions and new ideas._

_Have a new idea and want to discuss it? [Open a discussion]() so we can push things further._

File: app/views/site/contribute.erb (hunk 18 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: This is great! 👍

File: app/views/site/contribute.erb (hunk 12 above)
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Do it!

File: app/views/site/contribute.erb (hunk 17 above)
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Yeah, this is nice. Let's keep it.

**Comment by bernardoamc** [comment]
Ok, so this is the current state of the page. Some sections are bigger than others (which is not a problem imo), but should we group them by size or sort by some other criteria? 

screen shot 2016-05-30 at 10 55 14 pm
screen shot 2016-05-30 at 10 55 26 pm

**Comment by kytrinyx** [comment]
I like it. I think maybe we could give the two shortest one a bit more detail, but it doesn't really matter—let's just leave it all uneven for now.

I need to sleep before I review the final version, but so far this is looking great.

File: app/views/site/contribute.erb [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+  <section class="contributing-intro jumbotron">
+    <h1>Contribute to Exercism</h1>
+
+    <p class="lead">
+      <strong>So you want to contribute to exercism?</strong><br />
+      That is great news! We welcome all sorts of contributions and will do everything we can to help you get started.
+    </p>
+
+    <p>
+      <strong>Exercism is a project comprised of several repositories and these can be divided basically in:</strong>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: s/basically in/basically into/

File: app/views/site/contribute.erb [hunk 21]
```diff
@@ -0,0 +1,168 @@
...
+
+      <ol>
+        <li><strong>The website</strong>: Browse and review solutions, and have conversations.</li>
+        <li><strong>Command line interface</strong>: Used to fetch exercises and submit solutions.</li>
+        <li><strong>Language track repositories</strong>: Each language has its own repository with available problems.</li>
+        <li><strong>Problem metadata</strong>: Contain the language-specific implementations of exercises. Each language has its own repository.</li>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: s/Contain/Contains/

**Comment by bernardoamc** [comment]
After the review we can discuss where we want to add a link to this page. So far the footer seems like a good place, but I think we can benefit from adding it in other places (not sure where though).

File: app/views/site/contribute.erb [hunk 22]
```diff
@@ -0,0 +1,168 @@
...
+    <p class="lead">
+      <strong>So you want to contribute to exercism?</strong><br />
+      That is great news! We welcome all sorts of contributions and will do everything we can to help you get started.
+    </p>
+
+    <p>
+      <strong>Exercism is a project comprised of several repositories and these can be divided basically in:</strong>
+
+      <ol>
+        <li><strong>The website</strong>: Browse and review solutions, and have conversations.</li>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _Holds the code that people submit. Browse and review solutions, and have conversations._

File: app/views/site/contribute.erb [hunk 23]
```diff
@@ -0,0 +1,168 @@
...
+        <li><strong>Command line interface</strong>: Used to fetch exercises and submit solutions.</li>
+        <li><strong>Language track repositories</strong>: Each language has its own repository with available problems.</li>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: _Contain the exercise implementations for each language._

File: app/views/site/contribute.erb [hunk 24]
```diff
@@ -0,0 +1,168 @@
...
+        <li><strong>Problem metadata</strong>: Contain the language-specific implementations of exercises. Each language has its own repository.</li>
+        <li><strong>Problems API</strong>: Delivers exercises, consumed by the Exercism command-line client, as well as the Exercism website.</li>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: `s/Delivers exercises, c/Delivers exercises. C/`

File: app/views/site/contribute.erb [hunk 25]
```diff
@@ -0,0 +1,168 @@
...
+          <p>
+            Another great way to add new exercises is port an existing exercise to one of the other languages. This is easier than making up an exercise from scratch, since other people have thought through which test cases are needed.
+          </p>
+
+          <p>
+            Track maintainers are more than happy to give you all the support you need to successfully contribute.
+          </p>
+
+          <p>
+            Check out the detailed guide <a href="https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md#porting-an-exercise-to-another-language-track">here</a> if you want to give it a try!
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: We can drop the "here" if we make "detailed guide" the anchor text.

File: app/views/site/contribute.erb (hunk 23 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: s/Contain/Contains/  to keep the same pattern?

File: app/views/site/contribute.erb (hunk 25 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: 👍

File: app/views/site/contribute.erb [hunk 26]
```diff
@@ -0,0 +1,168 @@
...
+          <p>
+            We want the quality of the exercises to be as high as possible (and reviewing pull requests is a great way to both learn and teach a programming language).
+          </p>
+
+          <p>
+            We look at all sorts of things: style, idiomatic use of language features, naming, missing test cases, and redundant test cases. We also try to avoid forcing the solution to take a particular approach. Subscribe to the language track repositories of your favorite languages so you get notifications about new issues and pull requests. The track maintainers will greatly appreciate your help!
+          </p>
+
+          <p>
+             You can search for language track repositories <a href="https://github.com/exercism/">here</a>.
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Random thought... should we have a section at the bottom of the page that lists all the language track repositories?

File: app/views/site/contribute.erb [hunk 27]
```diff
@@ -0,0 +1,168 @@
...
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Help Triage Issues</h2>
+
+          <p>
+            We've got a huge number of old issues that aren't particularly actionable, and new issues come in all the time.
+          </p>
+
+          <p>
+            Is an issue no longer relevant? Can it be closed? Do we need more information? If it's a bug, is it reproducible? If it's a discussion and we need to make some sort of decision, then it should be labeled as such.
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: I think we need a final sentence in this paragraph, something like: _Help us get our issues under control._

File: app/views/site/contribute.erb [hunk 28]
```diff
@@ -0,0 +1,168 @@
...
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Fix Bugs</h2>
+
+          <p>
+            Exercism is currently a collection of experiments that grew into an app with the help of duct tape, chewing gum, and chicken wire. It's a mess, and it's got bugs.
+          </p>
+
+          <p>
+            Check out <a href="https://github.com/exercism/exercism.io/issues?q=is%3Aissue+is%3Aopen+label%3Abug">the current issues</a> if you want to give it a try!
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Maybe _Check out the [list of open bugs] if you want to help fix broken stuff.

File: app/views/site/contribute.erb [hunk 29]
```diff
@@ -0,0 +1,168 @@
...
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Add Features</h2>
+
+          <p>
+            Exercism is a collective work, we are always open to suggestions and new ideas.
+          </p>
+
+          <p>
+            Have a new idea and want to discuss it? <a href="https://github.com/exercism/discussions/issues">Open a discussion</a> so we can push things further.
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Here we have "discuss" and "discussions" right after each other. Maybe _Have a new idea? [Open a discussion] so we can push things further._

File: app/views/site/contribute.erb (hunk 28 above)
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Oh, wait. I like what I tried earlier:

_ Help us troubleshoot, find reproduction steps, and fix bugs._

File: app/views/site/contribute.erb (hunk 26 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: I don't know, usually people already know which languages are implemented (which made me think that we don't have a section about adding new languages). Do you think we will lose people in the search step?

File: app/views/site/contribute.erb (hunk 27 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: So far this is the only section without links, what do you think of directing people to major repository issues? I think they are `x-common` and `exercism.io`?

Comment (kytrinyx) [inline app/views/site/contribute.erb]: Yeah, let's direct them to https://github.com/exercism/exercism.io/issues.

The x-common repo is actually under control. There are things that I want to move into the discussions repository, but I've tagged them all. exercism.io is where it hurts.

File: app/views/site/contribute.erb (hunk 26 above)
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Oh! Yeah, we definitely should have a section about implementing new languages. I think we have a section in the x-common contributing guide about that.

We also should have a section about helping launch "in the works" or "planned" languages. These are the ones that are listed on the http://exercism.io/languages page in the lower sections.

I do think that it would be nice to be able to link people to something specific that is
language track -> repository

Maybe we can have a separate page like exercism.io/repositories that lists all the repositories (we can hand-code important ones and then generate the list of all the language track repositories.

Comment (kytrinyx) [inline app/views/site/contribute.erb]: I think this would be better than just pointing to the org on github, because there's a lot of messy legacy stuff there, as well as tools that they shouldn't be confronted with, and the sort order isn't configurable.

**Comment by kytrinyx** [comment]
Yeah, definitely in the footer, and the other place that immediately comes to mind is the dashboard.

File: app/views/site/contribute.erb (hunk 26 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: I like the idea of having a `exercism.io/repositories` page, this section would forward people to it. Should we open an issue as a reminder? This way we can start the page after we merge this one (or if anyone else is interested, they can already start the page layout).

Comment (kytrinyx) [inline app/views/site/contribute.erb]: Cool, yeah I'll open an issue now.

Comment (bernardoamc) [inline app/views/site/contribute.erb]: > Oh! Yeah, we definitely should have a section about implementing new languages. I think we have a section in the x-common contributing guide about that.

> We also should have a section about helping launch "in the works" or "planned" languages. These are the ones that are listed on the http://exercism.io/languages page in the lower sections.

Awesome, didn't know about this!

Comment (kytrinyx) [inline app/views/site/contribute.erb]: Done: https://github.com/exercism/exercism.io/issues/2924

File: app/views/site/contribute.erb [hunk 30]
```diff
@@ -0,0 +1,202 @@
...
+            If you like to think through usability issues, communication, design, features, roadmap, collaboration, or any other exciting open source stuff, join us in the <a href="https://github.com/exercism/discussions/issues">discussions repository</a>, where we're trying to sort out Exercism's future.
+          </p>
+        </div>
+      </div>
+    </div>
+
+    <div class="row">
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>New Languages</h2>
```
Comment (bernardoamc) [inline app/views/site/contribute.erb]: @kytrinyx I have added the two sections that you previously mentioned, the text is open for change (as everything else). :)

File: app/views/site/contribute.erb (hunk 26 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: That was fast! 🎉

File: app/views/site/contribute.erb [hunk 31]
```diff
@@ -0,0 +1,202 @@
...
+
+          <p>
+            Missing a language? We're always looking to increase our supported language tracks. Suggest it in the <a href="https://github.com/exercism/x-common">metadata repository</a>.
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Actually the instructions in the contributing guide are to email me at kytrinyx@exercism.io - let's use that here as well.

File: app/views/site/contribute.erb (hunk 30 above)
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Let's see if we can find an actionable header (so far we've been really good at headers that are essentially: **Do this!**).

Add New Languages?
Start a New Track?

File: app/views/site/contribute.erb [hunk 32]
```diff
@@ -0,0 +1,202 @@
...
+
+           <p>
+            Check out the section about <a href="https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md#starting-a-new-track">starting a new track</a> in the Contributing Guide.
+          </p>
+        </div>
+      </div>
+
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>In the Works</h2>
```
Comment (kytrinyx) [inline app/views/site/contribute.erb]: Again the actionable title thing. Damn, this is hard.

Comment (kytrinyx) [inline app/views/site/contribute.erb]: Help Launch WIP Tracks?

Then we need to explain WIP below.

Comment (bernardoamc) [inline app/views/site/contribute.erb]: This will not fit in a single line, `Launch WIP Tracks` will do the trick here? I don't know if it sounds misleading.

File: app/views/site/contribute.erb (hunk 30 above)
Comment (bernardoamc) [inline app/views/site/contribute.erb]: `Start a New Track` 👍

File: app/views/site/contribute.erb (hunk 32 above)
Comment (kytrinyx) [inline app/views/site/contribute.erb]: How about we just fold this contents into the "Add New Languages" and say that some tracks are in the works, so people can help launch them (and how to find them on the languages page)?

Comment (kytrinyx) [inline app/views/site/contribute.erb]: Help Launch a Track?

Comment (bernardoamc) [inline app/views/site/contribute.erb]: `Help Launch a Track` is good and makes sense to merge both sections using this title.

Comment (bernardoamc) [inline app/views/site/contribute.erb]: Now we have to explain both cases (new/lauch track) and refer to the contributing guide for each one.

File: app/views/site/contribute.erb [hunk 33]
```diff
@@ -0,0 +1,196 @@
...
+      </div>
+    </div>
+
+    <div class="row">
+      <div class="col-md-4">
+        <div class="contributing-section">
+          <h2>Help Launch a Track</h2>
+
+          <p>
+            Sometimes a language track is already planned or even exists, but are not in a state to be released just yet. These tracks can be seen in the <a href="http://exercism.io/languages">lower sections</a> of the languages page.
```
Comment (bernardoamc) [inline app/views/site/contribute.erb]: Merged the sections, it currently has all the necessary links (contributing sections, your email and the languages page), trying to improve the current section text.

File: app/views/application/_footer.erb [hunk 34]
```diff
@@ -10,6 +10,9 @@
         <p>
           <a href="/about">About</a>
         </p>
+        <p>
+          <a href="/contribute">Contribute to Exercism</a>
```
Comment (bernardoamc) [inline app/views/application/_footer.erb]: The current link is right below the `About` link in the footer, the alternative is adding it to `app/views/application/footer/links.erb`. I chose to place it under `About` because placing it under `The Command-Line Client` link felt strange (layout-wise). But since I'm far from being an expert, I really want a second opinion. 😄

File: app/views/site/contribute.erb (hunk 18 above)
Comment (machikoyasuda) [inline app/views/site/contribute.erb]: How about `Improve user experience` or `Improve UX`? The title `Design Stuff` makes it sound less important and a little magical & mysterious, `stuff`.

Comment (bernardoamc) [inline app/views/site/contribute.erb]: This is a great point, in no way we want people to think that improving the UX isn't important.

I like `Improve user experience` better than `Improve UX`, but it will not fit in a single line unfortunately. I don't know exactly why, but I don't like `Improve UX` so much, but if our current title passes the wrong impression I'm all about changing it. =]

**Comment by bernardoamc** [comment]
@machikoyasuda @kytrinyx What do you think?

screen shot 2016-06-02 at 3 39 26 pm

**Comment by kytrinyx** [comment]
Oh, wow, yes. "Improve UX" is waaay better than "Design Stuff". The "design stuff" thing reflects how fuzzy my understanding of design is, not the sense of importance that I give it.

**Comment by kytrinyx** [comment]
> find the page they're looking for

I think that instead of focusing on one particular "task" it would be nice to focus on the pain that they're avoiding. Maybe:

_Want to help people be less frustrated and more successful?_

File: app/views/application/_footer.erb (hunk 34 above)
Comment (kytrinyx) [inline app/views/application/_footer.erb]: I like it. It's really different from "how to use the site" which is what that other group of links is about.

**Comment by kytrinyx** [comment]
@bernardoamc When you're ready for a final review/merge, let me know! (And if you're waiting on me for something, would you mind letting me know that, too? I've been a bit swamped in the past few days and it sometimes means I miss things here).

**Comment by kytrinyx** [comment]
Hey! I made an issue that we can link to in the "user experience" section: https://github.com/exercism/discussions/issues/34

**Comment by bernardoamc** [comment]
@kytrinyx Sorry for the delay, just got back from a trip. I think we are ready for a final review/merge, the only thing "pending" is to decide if we want to change the dashboard in this PR or if we want to merge this and open new PRs linking specific pages to this guide.

**Comment by bernardoamc** [comment]
> Hey! I made an issue that we can link to in the "user experience" section: exercism/discussions#34

Nice! Let me change this! <3

**Comment by machikoyasuda** [comment]
@bernardoamc Love it :D

**Comment by kytrinyx** [comment]
I'm in the middle of onboarding at my new job. I will get to this when I can (at the latest early next week).

**Comment by bernardoamc** [comment]
> I'm in the middle of onboarding at my new job. I will get to this when I can (at the latest early next week).

No problem, take your time and enjoy this new challenge. <3

**Comment by kytrinyx** [comment]
It is deployed!

Thank you so much @bernardoamc!
